server/src/main/java/org/apache/druid/client/JsonParserIterator.java

```diff
@@ -146,6 +160,18 @@ private void init()
           objectCodec = jp.getCodec();
         }
       }
+      catch (TimeoutException ignored) {
```

Comment:
Hmm, maybe it would be worth to log `ignored.getMessage()` if it's not null?

Reply:
BTW, I think the below catch of `InterruptedException` should also throw `QueryInterruptedException`. I don't think it should be necessarily done in this pr though.